sql/core/src/main/scala/org/apache/spark/sql/execution/joins/BroadcastHashJoinExec.scala

```diff
@@ -471,6 +518,8 @@ case class BroadcastHashJoinExec(
          |    }
          |  }
          |}
+         |// special case for NullAwareAntiJoin, if anyNull in streamedRow, row should be dropped.
+         |${ if (isNullAwareAntiJoin) s"else { $found = true; }" else ""}
```

Comment:
why do we need the change? We return earlier if isNullAwareAntiJoin is true.

Reply:
> why do we need the change? We return earlier if isNullAwareAntiJoin is true.

yes we need this change, there are five code path where isNullAwareAntiJoin = true

1. inputEmpty = true => return all rows
2. hashedRelation anyNullKeyExists = true => return no rows
3. streamedRow is null, drop that row
4. streamedRow is not null, found match in hashedRelation, drop that row
5. streamedRow is not null, found no match in hashedRelation, keep that row

so that we have to iterate the streamedSide, check every row on condition 3~5.